src/finch/finch_assembly/interpreter.py

```diff
@@ -329,6 +329,40 @@ def my_func(*args_e):
                                 f"Unrecognized function definition: {func}"
                             )
                 return AssemblyInterpreterModule(self, kernels)
+            case asm.Print(args):
```

Comment:
I'm not sure why we need a separate case for variable. I think we can just evaluate each argument by calling `self` on it. You can expect that print will always have multiple arguments.